storage/gcloud/aio/storage/blob.py

```diff
@@ -73,23 +75,33 @@ async def get_signed_url(  # pylint: disable=too-many-locals
             raise ValueError("expiration time can't be longer than 604800 "
                              'seconds (7 days)')
 
-        iam_client = iam_client or IamClient(service_file=service_file,
-                                             token=token, session=session)
-
-        quoted_name = quote(self.name, safe='')
-        canonical_uri = f'/{self.bucket.name}/{quoted_name}'
+        quoted_name = quote(self.name, safe=b'/~')
+        canonical_uri = f'/{quoted_name}'
 
         datetime_now = datetime.datetime.utcnow()
         request_timestamp = datetime_now.strftime('%Y%m%dT%H%M%SZ')
         datestamp = datetime_now.strftime('%Y%m%d')
 
-        service_account_email = (service_account_email
-                                 or iam_client.service_account_email)
+        # TODO: support for GCE_METADATA and AUTHORIZED_USER tokens. It should
+        # be possible via API, but seems to not be working for us in some
+        # cases.
+        #
+        # https://cloud.google.com/iam/docs/reference/credentials/rest/v1/
+        # projects.serviceAccounts/signBlob
+        token = token or self.bucket.storage.token
+        client_email = token.service_data.get('client_email')
+        key = token.service_data.get('private_key')
```

Comment:
Oh cool — does this mean that the current service account is sufficient to perform signing, as long as it was granted permission to sign?

Reply:
Yup! The docs say your token needs the `urlSign` permissions and the associated `SCOPES` granted, but in practice I was able to get this to work just fine with a token with only `storage.read` and the standard scopes associated with `glcoud-aio-storage`'s token. I left the `token` argument in this method to let folks pass in a specific urlsign token if need be, but it seems unnecessary... that said, rather avoid blocking anyone if I can help it!